### DISCUSSION TITLE
Systems with a generic `SystemParam` type

### DISCUSSION THREAD

**Discussion by Adamkob12** [bevyengine/bevy#11203] [Q&A] [answered]
I wasn't sure if this is a bug, and even if it is, whether it's an issue.

What is the expected behavior of this?:

<img width="644" alt="Screenshot 2024-01-03 at 20 06 10" src="https://github.com/bevyengine/bevy/assets/46227443/506502c6-64aa-41bb-8640-4912ee09f5b2">

Because it doesn't compile, I know this doesn't seem like something anyone would want to do, but, I'm pretty curious why it doesn't work, or how to fix it while keeping the system generic over SystemParam T.

Here's the error:

```Rust
the trait bound `fn(bevy::prelude::Query<'_, '_, bevy::prelude::Entity>) {generic_system::<bevy::prelude::Query<'_, '_, bevy::prelud
e::Entity>>}: bevy::prelude::IntoSystem<(), (), _>` is not satisfied
   --> main.rs:57:29
    |
57  |     app.add_systems(Update, generic_system::<Query<Entity>>);
    |         -----------         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `bevy::prelude::IntoSystem<(), (), _>` is not implemented for fn item 
`fn(bevy::prelude::Query<'_, '_, bevy::prelude::Entity>) {generic_system::<bevy::prelude::Query<'_, '_, bevy::prelude::Entity>>}`
    |         |
    |         required by a bound introduced by this call
```

**Comment by hymm** [comment, marked as answer]
been working on an improved example that shows how to do this: https://github.com/hymm/bevy/commit/25150b86b921b9385237452511ac34ba5d6dc865#diff-a395bbe0429c7fa43f10035ba3bcd5e0ca65820ff9648004ac1476e9ff41f7d6R93-R127

**Reply by Adamkob12** [reply]
Cool, thanks!

**Reply by VVishion** [reply]
Wrapping it with `ParamSet` is also possible.
